sdk/storage/inc/blobs/blob_client.hpp

```diff
@@ -12,6 +12,11 @@
 #include <memory>
 #include <string>
 
+namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+  class DirectoryClient;
+  class FileClient;
+}}}} // namespace Azure::Storage::Files::DataLake
+
```

Comment:
Do you think it'll look nicer if we move other forward declarations here?

```
namespace Azure { namespace Storage {

  namespace Files { namespace DataLake {
    class DirectoryClient;
    class FileClient;
  }} // namespace Files::DataLake

  namespace Blobs {
    class BlockBlobClient;
    class AppendBlobClient;
    class PageBlobClient;
  } // namespace Blobs

}} // namespace Azure::Storage
```

Reply:
I don't have a strong preference here. What I wrote sets a clearer boundary for different services, in case we want to remove it in the future it is easier. What you suggest is more concise in code. I can change if you insist.